packages/block-library/src/image/use-max-width-observer.js

```diff
@@ -1,17 +1,20 @@
 /**
  * WordPress dependencies
  */
-import { useRef } from '@wordpress/element';
+import { useState } from '@wordpress/element';
 import { useResizeObserver } from '@wordpress/compose';
 
-function useMaxWidthObserver() {
-	const [ contentResizeListener, { width } ] = useResizeObserver();
-	const observerRef = useRef();
+function useMaxWidthObserver( className ) {
+	const [ usedMaxWidth, setUsedMaxWidth ] = useState();
+	const setResizeObserved = useResizeObserver( ( [ entry ] ) => {
+		setUsedMaxWidth( entry.contentRect.width );
+	} );
 
 	const maxWidthObserver = (
 		<div
-			// Some themes set max-width on blocks.
-			className="wp-block"
+			// The block’s class names need to be applied because the max-width
+			// may vary per theme styles (e.g. alignments).
+			className={ className }
```

Comment:
The crucial change. As is, this puts some seemingly irrelevant classes onto the element e.g. (`block-editor-block-list__block` `is-selected`) but they don’t seem problematic from testing. We could probably filter these to just ones that start with `align` or `wp-block` though.

I’d prefer getting the measurement without inserting an extra element and have experimented with that. I wanted to try more minimal changes for the short term though.

Reply:
I filtered the applied class names down as mentioned in the above comment so I'm marking this resolved.